src/main/java/view/ViewBuilder.java

```diff
@@ -12,13 +14,13 @@
 public class ViewBuilder extends JPanel {
 
     private final String viewName;
-    private final List<JComponent> components;
+    private final Map<String, JButton> buttonsMap;
     private final JPanel buttonsPanel;
     private final JPanel mainPanel;
 
     public ViewBuilder(String viewName) {
```

Comment:
I don't think we should ever pass parameters through the constructor because the whole point of the Builder pattern is to decouple the constructor from the actual object.

Reply:
I did this myself, just approve my pull request or smth.